auction-server/src/auction/service/verification.rs

```diff
@@ -2752,6 +2753,85 @@ mod tests {
         }
     }
 
+    #[tokio::test]
+    async fn test_verify_bid_when_create_ata_relayer_payer() {
+        let (service, opportunities) = get_service(true);
+        let opportunity = opportunities.user_token_specified.clone();
+        let bid_amount = 1;
+        let searcher = Keypair::new();
+        let swap_instruction = svm::Svm::get_swap_instruction(GetSwapInstructionParams {
+            searcher: searcher.pubkey(),
+            opportunity_params: get_opportunity_params(opportunity.clone()),
+            bid_amount,
+            deadline: (OffsetDateTime::now_utc() + Duration::seconds(30)).unix_timestamp(),
+            fee_receiver_relayer: Pubkey::new_unique(),
+            relayer_signer: service.config.chain_config.express_relay.relayer.pubkey(),
+        })
+        .unwrap();
+        let relayer = service.config.chain_config.express_relay.relayer.pubkey();
+        let program = match opportunity.program.clone() {
+            OpportunitySvmProgram::Swap(program) => program,
+            _ => panic!("Expected swap program"),
+        };
+        let create_ata_instruction =
+            spl_associated_token_account::instruction::create_associated_token_account(
+                &relayer,
+                &relayer,
+                &opportunity.buy_tokens[0].token,
+                &program.token_program_user,
+            );
+        let result = get_verify_bid_result(
+            service.clone(),
+            searcher.insecure_clone(),
+            vec![create_ata_instruction, swap_instruction.clone()],
+            opportunities.user_token_specified.clone(),
+        )
+        .await;
+        assert_eq!(
+            result.unwrap_err(),
+            RestError::InvalidInstruction(
+                Some(0),
+                InstructionError::InvalidPayerInCreateAtaInstruction {
+                    expected: searcher.pubkey(),
+                    found:    relayer,
+                }
+            )
+        );
+    }
+
+    #[tokio::test]
+    async fn test_verify_bid_when_sol_transfer_relayer() {
```

Comment:
I think `test_verify_bid_when_relayer_transfer_instruction` already covered this case